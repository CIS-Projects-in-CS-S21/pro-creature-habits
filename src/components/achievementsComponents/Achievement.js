import {useSelector} from "react-redux";
import {Image, Text, View, StyleSheet} from "react-native";
import * as Progress from "react-native-progress";
import React from "react";
import { Ionicons } from "@expo/vector-icons";

const styles = StyleSheet.create({
	mainAchievementContainer: {
		backgroundColor: '#402688',
		padding: 10,
		paddingBottom: 15,
		borderTopLeftRadius: 10,
		borderBottomLeftRadius: 10,
		borderRightColor: 'black',
		borderRightWidth: 1,
		flexGrow: 1
	},
	achievementContainer: {
		flexDirection: 'row',
		borderRadius: 10,
		justifyContent: 'space-between',
		borderColor: 'black',
		shadowOffset: {width: 0, height: 5},
		shadowOpacity: 0.8,
		shadowRadius: 3,
		elevation: 11,
		marginBottom: 15,
	},
	achievementReward: {
		flexDirection: 'row',
		width: 20,
		justifyContent: 'center',
		backgroundColor: '#3156c4',
		borderTopRightRadius: 10,
		borderBottomRightRadius: 10,
		flexBasis: '20%',
		alignItems: 'center'
	},
	achievementText: {
		color: 'white',
		fontSize: 16,
		paddingLeft: 10
	},
	textProgress: {
		color: 'white',
		fontSize: 16,
		paddingRight: 10
	},
	achievementHeader: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		paddingTop: 5
	},
	achievementRewardText: {
		color: 'white',
		textAlign: 'center',
		fontSize: 20
	}
})

const Achievement = ({achievement}) => {
	const ach = useSelector(state=>state.achievements);
	const filter = useSelector(state=>state.achievementsFilter);

	return (
		(ach[achievement][filter].complete && ach[achievement][filter].reward_taken ? (
			<View style={styles.achievementContainer}>
				<View style={[styles.mainAchievementContainer, {backgroundColor: '#636363', borderRightColor: 'black'}]}>
					<View style={styles.achievementHeader}>
						<Text style={styles.achievementText}>{ach[achievement].name}</Text>
						<Text style={styles.textProgress}>{`${ach[achievement][filter].to_complete}/${ach[achievement][filter].to_complete}`}</Text>
					</View>
					<Progress.Bar style={{margin: 10}} progress={1} width={null} height={10} color='white' borderColor='white' />
				</View>
				<View style={[styles.achievementReward, {backgroundColor: '#868686'}]}>
					<Ionicons name="checkmark" color="white" size={22}/>
				</View>
			</View>
		) : ach[achievement][filter].complete ? (
			<View style={styles.achievementContainer}>
				<View style={[styles.mainAchievementContainer, {backgroundColor: '#4a992c', borderRightColor: 'white'}]}>
					<View style={styles.achievementHeader}>
						<Text style={styles.achievementText}>{ach[achievement].name}</Text>
						<Text style={styles.textProgress}>{`${ach[achievement][filter].to_complete}/${ach[achievement][filter].to_complete}`}</Text>
					</View>
					<Progress.Bar style={{margin: 10}} progress={1} width={null} height={10} color='white' borderColor='white' />
				</View>
				<View style={[styles.achievementReward, {backgroundColor: '#4ab125'}]}>
					<Text style={styles.achievementRewardText}>{ach[achievement][filter].reward}</Text>
					<Image style={{width: 20, height: 20}} source={require('../../test_images/coin.png')}/>
				</View>
			</View>
		) : (
			<View style={styles.achievementContainer}>
				<View style={styles.mainAchievementContainer}>
					<View style={styles.achievementHeader}>
						<Text style={styles.achievementText}>{ach[achievement].name}</Text>
						<Text style={styles.textProgress}>{`${ach[achievement].completed}/${ach[achievement][filter].to_complete}`}</Text>
					</View>
					<Progress.Bar style={{margin: 10}} progress={ach[achievement].completed/ach[achievement][filter].to_complete} width={null} height={10} color='#4ab125' borderColor='white' />
				</View>
				<View style={styles.achievementReward}>
					<Text style={styles.achievementRewardText}>{ach[achievement][filter].reward}</Text>
					<Image style={{width: 20, height: 20}} source={require('../../test_images/coin.png')}/>
				</View>
			</View>
		))
	)
}

export default Achievement;
