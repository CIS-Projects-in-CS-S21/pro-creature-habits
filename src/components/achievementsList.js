const achievementsList = {
	buy_item: {
		name: 'Buy item(s)',
		completed: 0,
		beginner: {
			to_complete: 1,
			reward: 5,
			complete: false,
			reward_taken: false
		},
		intermediate: {
			to_complete: 20,
			reward: 50,
			complete: false,
			reward_taken: false
		},
		hard: {
			to_complete: 100,
			reward: 250,
			complete: false,
			reward_taken: false
		}
	},
	complete_task: {
		name: 'Complete task(s)',
		completed: 0,
		beginner: {
			to_complete: 1,
			reward: 5,
			complete: false,
			reward_taken: false
		},
		intermediate: {
			to_complete: 20,
			reward: 50,
			complete: false,
			reward_taken: false
		},
		hard: {
			to_complete: 100,
			reward: 250,
			complete: false,
			reward_taken: false
		}
	},
	buy_food: {
		name: 'Buy food',
		completed: 0,
		beginner: {
			to_complete: 1,
			reward: 5,
			complete: false,
			reward_taken: false
		},
		intermediate: {
			to_complete: 20,
			reward: 50,
			complete: false,
			reward_taken: false
		},
		hard: {
			to_complete: 100,
			reward: 250,
			complete: false,
			reward_taken: false
		}
	},
	buy_clothes: {
		name: 'Buy clothes',
		completed: 0,
		beginner: {
			to_complete: 1,
			reward: 5,
			complete: false,
			reward_taken: false
		},
		intermediate: {
			to_complete: 10,
			reward: 50,
			complete: false,
			reward_taken: false
		},
		hard: {
			to_complete: 40,
			reward: 250,
			complete: false
		}
	},
	buy_toy: {
		name: 'Buy toy(s)',
		completed: 0,
		beginner: {
			to_complete: 1,
			reward: 5,
			complete: false,
			reward_taken: false
		},
		intermediate: {
			to_complete: 10,
			reward: 50,
			complete: false,
			reward_taken: false
		},
		hard: {
			to_complete: 40,
			reward: 250,
			complete: false,
			reward_taken: false
		}
	},
};

export {achievementsList}
