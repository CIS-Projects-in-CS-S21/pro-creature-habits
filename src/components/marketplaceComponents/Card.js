import React from "react";
import {Image, Text, View, StyleSheet} from "react-native";
import {ItemInventory} from "../ItemInventory";
import {useSelector} from "react-redux";
import {Ionicons} from "@expo/vector-icons";

const styles = StyleSheet.create({
	cardContainer: {
		borderWidth: 3,
		borderColor: 'white',
		padding: 12,
		backgroundColor: 'rgba(49,69,194,0.7)',
		shadowOffset: {width: 0, height: 1},
		shadowOpacity: 0.8,
		shadowRadius: 3,
		elevation: 11,
		borderRadius: 5
	},
	cardFooter: {
		flexDirection: 'row',
		justifyContent: 'space-evenly',
		paddingLeft: 5,
		paddingRight: 5,
		marginTop: 5,
		borderTopWidth: 2,
		borderTopColor: 'darkgrey'
	},
	cardFooterText: {
		fontSize: 20,
		color: 'white',
		marginTop: 5
	},
	coin: {
		width: 25,
		height: 25,
		marginTop: 5
	},
	itemImage: {
		width:75,
		height:75,
		shadowOffset: {width: 0, height: 1},
		shadowOpacity: 0.8,
		shadowRadius: 3,
	}
});

const Card = ({item}) => {
	const itemsBought = useSelector(state=>state.itemsBought);

	return (
		<View>
			{!itemsBought.includes(item) ? (
				<View style={styles.cardContainer}>
					<Image style={styles.itemImage} source={ItemInventory[item].uri}/>
					<View style={styles.cardFooter}>
						<Text style={styles.cardFooterText}>{ItemInventory[item].cost}</Text>
						<Image style={styles.coin} source={require('../../test_images/coin.png')}/>
					</View>
				</View>
			) : (
				<View style={[styles.cardContainer, {backgroundColor: '#2a2a2a'}]}>
					<Image style={styles.itemImage} source={ItemInventory[item].uri}/>
					<View style={styles.cardFooter}>
						<Ionicons name="checkmark" color="white" size={28}/>
					</View>
				</View>
			)}
		</View>
	)
}

export default Card;
