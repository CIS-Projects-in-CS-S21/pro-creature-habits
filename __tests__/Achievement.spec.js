import * as React from 'react';
import renderer from 'react-test-renderer';
import Achievement from '../src/components/achievementsComponents/Achievement';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider , createStore, combineReducers} from 'react-redux';
import configureStore from 'redux-mock-store'
import achievementsList from '../src/components/achievementsList';
import achievementsFilter from '../src/components/filterComponents/AchievementsFilter';
import marketplaceFilter from '../src/components/filterComponents/MarketplaceFilter';
import petInventoryFilter from '../src/components/filterComponents/PetInventoryFilter';
import Card from '../src/components/marketplaceComponents/Card'
enzyme.configure({ adapter: new Adapter() });
let store;
import marketplaceItemsBoughtReducer from '../src/redux/marketplaceItemsBought'
const createTestStore = () => {
  const store = createStore(
    combineReducers({
      itemsBought: marketplaceItemsBoughtReducer,
    })
  );
  return store;
}
describe('<Achievement />', () => {
    const mockStore = configureStore();
        let store, wrapper;


    it("achievement not null", () => {

        expect(Achievement).toBeTruthy();
    })

    it("achievement Filter not null", () => {

            expect(achievementsFilter).toBeTruthy();
        })



it('achievement Filter snapshot renders',()=>{
    //store = mockStore(achievementsList)
        const achievementComponent = renderer.create(<achievementsFilter />).toJSON()
        expect(achievementComponent).toMatchSnapshot();
    });

    it("achievement Filter snapshot matches tree", () => {

            // store = mockStore(initialState)
            const comp = <Provider><achievementsFilter/></Provider>
            const tree = renderer.create().toJSON(comp);

            expect(tree).toMatchSnapshot();
        });

        it('marketplace Filter snapshot renders',()=>{
            //store = mockStore(achievementsList)
                const marketplaceComponent = renderer.create(<marketplaceFilter />).toJSON()
                expect(marketplaceComponent).toMatchSnapshot();
            });

            it("marketplace Filter snapshot matches tree", () => {

                    // store = mockStore(initialState)
                    const comp = <Provider><marketplaceFilter/></Provider>
                    const tree = renderer.create().toJSON(comp);

                    expect(tree).toMatchSnapshot();
                });

             it('petInventoryFilter snapshot renders',()=>{
                         //store = mockStore(achievementsList)
                             const petInventoryComponent = renderer.create(<petInventoryFilter />).toJSON()
                             expect(petInventoryComponent).toMatchSnapshot();
                         });

                         it("petInventory Filter snapshot matches tree", () => {

                                 // store = mockStore(initialState)
                                 const comp = <Provider><petInventoryFilter/></Provider>
                                 const tree = renderer.create().toJSON(comp);

                                 expect(tree).toMatchSnapshot();
                             });

})