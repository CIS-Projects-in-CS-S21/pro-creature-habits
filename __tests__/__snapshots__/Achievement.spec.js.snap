// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Achievement /> achievement Filter snapshot matches tree 1`] = `null`;

exports[`<Achievement /> achievement Filter snapshot renders 1`] = `<achievementsFilter />`;

exports[`<Achievement /> marketplace Filter snapshot matches tree 1`] = `null`;

exports[`<Achievement /> marketplace Filter snapshot renders 1`] = `<marketplaceFilter />`;

exports[`<Achievement /> petInventory Filter snapshot matches tree 1`] = `null`;

exports[`<Achievement /> petInventoryFilter snapshot renders 1`] = `<petInventoryFilter />`;
